packages/relay-compiler/transforms/RelayMaskTransform.js

```diff
@@ -149,7 +148,6 @@ function areSameArgumentDefinitions(
   return (
     argDef1.kind === argDef2.kind &&
     argDef1.name === argDef2.name &&
-    isEquivalentType(argDef1.type, argDef2.type) &&
```

Comment:
hey @robrichard I think that instead of entirely removing this check, would it make sense to instead check that the input type is a suptype of the expected argument? i.e. in your example in #2358 that $myReturn is a subtype of `return:`.

Does that make sense?